src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] dataArray;
+    private int size;
+
+    public ArrayList() {
+        dataArray = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growUp();
+        dataArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        growUp();
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index < size && index >= 0) {
+            System.arraycopy(dataArray, index, dataArray, index + 1, size - index);
+            dataArray[index] = value;
+            size++;
+            return;
+        }
+        throw new ArrayListIndexOutOfBoundsException(
+                "Index " + index + " is wrong for size " + size);
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() > size) {
+            growUp();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            dataArray[size] = list.get(i);
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validationForIndex(index);
+        return (T)dataArray[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        validationForIndex(index);
+        growUp();
```

Comment:
not sure if here need that method, cause here we are just setting new value to existing, and if our index is out of bounds than this method won't work anyway. nut if it fits our size, we don't need to grow it more.(maybe this code works with your solutions, check it, or maybe you don't need it here) ^_^